### PR TITLE
fix: teleport page - balance not updated

### LINKED ISSUE #7821: teleport page - balance not updated
in teleport page
need to open wallet modal in order to get balance to update

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

  👇 __ Let's make a quick check before the contribution.

  ## PR Type

  - [x] Bugfix
  - [x] Refactoring

  ## Needs QA check

  - @kodadot/qa-guild please review

  ## Context

  - [x] Closes #7821
  - [ ] Requires deployment <snek/rubick/worker>

  #### Did your issue had any of the "$" label on it?

  - [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)

  ## Screenshot 📸

  - [ ] My fix has changed UI

  ## Copilot Summary
  
### <samp>🤖 Generated by Copilot at c1aa829</samp>

Refactored the code for displaying multiple balances across chains and tokens by creating a new composable function `useMultipleBalance` and using it in `Teleport.vue` and `MultipleBalances.vue`. The new function simplifies the component code and improves the performance and reliability of the balance queries.

  
### <samp>🤖 Generated by Copilot at c1aa829</samp>

> _Sing, O Muse, of the skillful coder who devised_
> _A composable function, `useMultipleBalance`, to obtain_
> _The wealth of many chains and tokens, swiftly fetched and updated_
> _By the grace of `useIdentityStore` and `useAuth`, and other aids._

### CODE REVIEW

File: composables/useMultipleBalance.ts [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    multiBalances,
+    multiBalanceAssets,
+    multiBalanceAssetsTestnet,
+    multiBalanceNetwork,
+  } = storeToRefs(identityStore)
+
+  const currentNetwork = computed(() =>
+    isTestnet.value ? 'test-network' : 'main-network',
+  )
+  function calculateUsd(amount: string, token = 'KSM') {
```
Comment (roiLeo) [inline composables/useMultipleBalance.ts:40]:
```suggestion
  const calculateUsd = (amount: string, token = 'KSM') => {
```

File: composables/useMultipleBalance.ts [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+    const amountToNumber = Number(amount.replace(/\,/g, ''))
+
+    return calculateExactUsdFromToken(
+      amountToNumber,
+      Number(fiatStore.getCurrentTokenValue(token)),
+    )
+  }
+
+  async function getBalance(chainName: string, token = 'KSM', tokenId = 0) {
```
Comment (roiLeo) [inline composables/useMultipleBalance.ts:53]:
```suggestion
  const getBalance = async (chainName: string, token = 'KSM', tokenId = 0) => {
```

File: composables/useMultipleBalance.ts [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+  onMounted(async () => {
+    if (currentNetwork.value !== multiBalanceNetwork.value) {
+      identityStore.resetMultipleBalances()
+    }
+
+    fetchMultipleBalance()
+    refetchMultipleBalanceTimer.value = setInterval(() => {
+      fetchMultipleBalance()
+    }, 30000)
```
Comment (roiLeo) [inline composables/useMultipleBalance.ts:130]: what's this?

**Review by roiLeo — APPROVED** [review]
otherwise lgtm

File: composables/useMultipleBalance.ts (hunk 3 above)
Comment (Jarsen136) [reply composables/useMultipleBalance.ts:130]: It updates the user's balance on different chains every 30 seconds.

Comment (roiLeo) [reply composables/useMultipleBalance.ts:130]: Won't it be resource-hungry? what happen if wss endpoint is dead/off?

Comment (Jarsen136) [reply composables/useMultipleBalance.ts:130]: > Won't it be resource-hungry?

Without this update, the balance would not change anymore after first fetching. We introduced this regular update to solve the previous issue about incorrect balance.

>  what happen if wss endpoint is dead/off?

Then the balance from the specific endpoint, would not displayed of course.

Comment (roiLeo) [reply composables/useMultipleBalance.ts:130]: > Then the balance from the specific endpoint, would not displayed of course.

without blocking others?

Comment (Jarsen136) [reply composables/useMultipleBalance.ts:130]: > > Then the balance from the specific endpoint, would not displayed of course.
> 
> without blocking others?

No. It's the same as what we do in the production environment.

Comment (roiLeo) [reply composables/useMultipleBalance.ts:130]: thank you for these clarifications

**Comment by yangwao** [comment]
Thanks!
pay 30 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $30 @ 4.35 USD/DOT ~ 6.897 $DOT
🧗 [16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z](https://polkadot.subscan.io/account/16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)
🔗 [0x785f01fa5e6b9523f11861088d94be144a88d39d033c560c534c2905e650f809](https://polkadot.subscan.io/extrinsic/0x785f01fa5e6b9523f11861088d94be144a88d39d033c560c534c2905e650f809)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**